lib/msal-browser/docs/testing.md

```diff
@@ -2,114 +2,129 @@
 
-**Note: This is an advanced feature that is intended for testing purposes in the browser environment only. Loading tokens to your application's cache may cause your app to break. Additionally, we recommend `loadExternalTokens()` API to be used with unit and integration tests. For E2E testing, please refer to our [TestingSample](../../../samples/msal-browser-samples/TestingSample) instead.**
+**Note: This is an advanced feature that is intended for testing purposes in the browser environment only. We do not recommend using this in a production app. For E2E testing recommendations, please refer to our [TestingSample](../../../samples/msal-browser-samples/TestingSample) instead.**
 
-The `loadExternalTokens()` API can be accessed by calling `getTokenCache()` on MSAL Browser's `PublicClientApplication` instance. 
+The `loadExternalTokens()` API can be accessed by calling `getTokenCache()` on MSAL Browser's `PublicClientApplication` instance.
 
 ```js
 const msalTokenCache = myMSALObj.getTokenCache();
-msalTokenCache.loadExternalTokens(silentRequest, serverResponse, loadTokenOptions);
+msalTokenCache.loadExternalTokens(
+    silentRequest,
+    serverResponse,
+    loadTokenOptions
+);
 ```
 
 `loadExternalTokens()` takes in a request of type `SilentRequest`, a response of type `ExternalTokenResponse`, and options of type `LoadTokenOptions`.
 
 See the type definitions for each, which can be imported from `@azure/msal-browser`:
 
-- [`SilentRequest`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/modules/_azure_msal_browser.html#silentrequest)
-- [`ExternalTokenResponse`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/modules/_azure_msal_common.html#externaltokenresponse)
-    - Note that the server response you receive will also have a refresh token attached. Currently, `loadExternalTokens()` does not load refresh tokens.
+-   [`SilentRequest`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/types/_azure_msal_browser.SilentRequest.html)
+-   [`ExternalTokenResponse`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/types/_azure_msal_browser.ExternalTokenResponse.html)
+-   [`LoadTokenOptions`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/types/_azure_msal_browser.LoadTokenOptions.html)
 
-```ts
-export type LoadTokenOptions = {
-    clientInfo?: string,
-    extendedExpiresOn?: number
-};
-```
+## Loading tokens
+
+You can provide any combination of id, access and refresh tokens for caching but at a minimum the `loadExternalTokens` API requires one of the following sets of input parameters to identify who the tokens belong to and be capable of storing them in the cache:
+
+-   A `SilentRequest` object with account information, OR
+-   A `SilentRequest` object with the authority AND a `LoadTokenOptions` object with `clientInfo`, OR
+-   A `SilentRequest` object with the authority AND a server response object with `client_info`
```

Comment:
Do we need to add reference to use cases where `client_info` is not available? I understand it is dependent on the Confidential client flow that precedes the side loading, it may help for comprehensive use case reference in this case.

Reply:
The line below this one mentions you can use idToken instead and 2 lines above mentions you can use AccountInfo instead.

Reply:
I meant something else entirely. More about "If you are ADFS, you will not have ` client_info` etc" .. but on second thought, folks who use this feature are already not doing simple use cases. Lets leave this as is.